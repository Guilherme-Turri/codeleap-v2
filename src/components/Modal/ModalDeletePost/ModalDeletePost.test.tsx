import { ModalDeletePost } from '.'
import { componentRender } from '../../../render/componentRender'
import { screen, fireEvent } from '@testing-library/react'
describe('<ModalDeletePost />', () => {
  const modalProps = {
    handleSuccess: jest.fn(),
    handleCancel: jest.fn(),
    handleDelete: jest.fn(),
    message: 'Message displayed'
  }
  it('should render on screen with message and action suceess', () => {
    const { container } = componentRender(<ModalDeletePost {...modalProps} />)
    expect(screen.getByText(modalProps.message)).toBeInTheDocument();
    const succesBtn = screen.getByTestId("successbtn");
    expect(succesBtn).toBeInTheDocument();
    fireEvent.click(succesBtn);
    expect(modalProps.handleSuccess).toHaveBeenCalled();
    expect(container).toMatchSnapshot();


  })
  it('should render on screen and Cancel Action', () => {
    modalProps.message = '';
    componentRender(<ModalDeletePost {...modalProps} />)
    expect(screen.getByText('Are you sure to delete this post?')).toBeInTheDocument();
    const cancelBtn = screen.getByTestId("cancelbtn");
    expect(cancelBtn).toBeInTheDocument();
    fireEvent.click(cancelBtn);
    expect(modalProps.handleCancel).toHaveBeenCalled();
  })
  it('should render on screen and do Delete Action', () => {
    modalProps.message = '';
    componentRender(<ModalDeletePost {...modalProps} />)
    expect(screen.getByText('Are you sure to delete this post?')).toBeInTheDocument();
    const deletebtn = screen.getByTestId("deletebtn");
    expect(deletebtn).toBeInTheDocument();
    fireEvent.click(deletebtn);
    expect(modalProps.handleDelete).toHaveBeenCalled();
  })

})