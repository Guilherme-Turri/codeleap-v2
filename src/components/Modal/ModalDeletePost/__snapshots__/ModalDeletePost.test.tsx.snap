// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ModalDeletePost /> should render on screen with message and action suceess 1`] = `
.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  height: 100%;
}

.c0 .c2 {
  font-size: 1.5rem;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 6rem;
  height: 3rem;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
}

.c5 {
  all: unset;
  background-color: #00CBA9;
  border-radius: 0.313rem;
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <p
        class="c2 c3"
      >
        Message displayed
      </p>
      <div
        class="c4"
        data-testid="successbtn"
      >
        <button
          class="c5"
        >
          OK
        </button>
      </div>
    </div>
  </div>
</div>
`;
